apps/admin-portal/src/components/identity-providers/wizard/identity-provider-create-wizard.tsx

```diff
@@ -329,49 +329,121 @@ export const IdentityProviderCreateWizard: FunctionComponent<IdentityProviderCre
-        const availableAuthenticator = _.find(availableAuthenticators, {
-            authenticatorId: template
-                .federatedAuthenticators?.authenticators[0]?.authenticatorId
-        });
-        if (availableAuthenticator) {
-            getAuthenticatorMetadata(availableAuthenticator.authenticatorId);
+        if (availableAuthenticators?.find(eachAuthenticator => eachAuthenticator.authenticatorId ===
+            template?.federatedAuthenticators?.defaultAuthenticatorId)) {
+            getAuthenticatorMetadata(template?.federatedAuthenticators?.defaultAuthenticatorId);
         }
     }, [availableAuthenticators]);
 
-    const getAuthenticatorProperties = () => {
-        return authenticatorMetadata?.properties.map(
+    /**
+     * Validate and get federate authenticators.
+     */
+    const getValidatedAuthenticators = () => {
+        const defaultAuthenticatorPropertiesFromMetadata = defaultAuthenticatorMetadata?.properties.map(
             (eachProp): AuthenticatorProperty => {
                 return {
                     key: eachProp?.key,
                     value: eachProp?.defaultValue
                 }
             });
+
+        // For the default authenticator, update values of it's properties with the corresponding value from the
+        // template, if any.
+        // todo Need to do the same for rest of the configured authenticators in the template.
+        const authenticatorsInTemplate = template?.federatedAuthenticators.authenticators;
+        return {
+            authenticators: authenticatorsInTemplate.map((authenticator) => {
+                return authenticator.authenticatorId === template.federatedAuthenticators.defaultAuthenticatorId ?
+                    {
+                        ...authenticator,
+                        properties: _.merge(defaultAuthenticatorPropertiesFromMetadata, authenticator.properties)
+                    } : authenticator;
+            }),
+            defaultAuthenticatorId: template.federatedAuthenticators.defaultAuthenticatorId
+        };
+    };
+
+    const isAuthenticatorStepAvailable = () => {
+        return template?.federatedAuthenticators?.defaultAuthenticatorId;
+    };
+
+    const getWizardSteps = () => {
+        let STEPS: WizardStepInterface[] = [
+            {
+                icon: IdentityProviderWizardStepIcons.general,
+                name: WizardSteps.GENERAL_DETAILS,
+                submitCallback: setSubmitGeneralSettings,
+                title: "General settings"
+            }
+        ];
+
+        if (isAuthenticatorStepAvailable()) {
+            STEPS = [
+                ...STEPS,
+                {
+                    icon: IdentityProviderWizardStepIcons.authenticator,
+                    name: WizardSteps.AUTHENTICATOR,
+                    submitCallback: setSubmitAuthenticator,
+                    title: "Authenticator"
+                }
+            ];
+
```

Comment:
Shall we remove this line?

Reply:
Fixed in 6311ed7b87d8256e38389b5145f845d46e4c8564.